pkg/controller/controller_instance.go

```diff
@@ -46,6 +46,54 @@ func (c *controller) instanceAdd(obj interface{}) {
 	c.instanceQueue.Add(key)
 }
 
+// Async operations on instances have a somewhat convoluted flow in order to
+// ensure that only a single goroutine works on an instance at any given time.
+// The flow is:
+//
+// 1.  When the controller wants to begin polling the state of an operation on
+//     an instance, it calls its beginPollingInstance method (or
+//     calls continuePollingInstance, an alias of that method)
+// 2.  begin/continuePollingInstance do a rate-limited add to the polling queue
+// 3.  the pollingQueue calls requeueInstanceForPoll, which adds the instance's
+//     key to the instance work queue
+// 4.  the worker servicing the instance polling queue forgets the instances key,
+//     requiring the controller to call continuePollingInstance if additional
+//     work is needed.
+// 5.  the instance work queue is the single work queue that actually services
+//     instances by calling reconcileInstance
+
+// requeueInstanceForPoll adds the given instance key to the controller's work
+// queue for instances.  It is used to trigger polling for the status of an
+// async operation on and instance and is called by the worker servicing the
+// instance polling queue.  After requeueInstanceForPoll exits, the worker
+// forgets the key from the polling queue, so the controller must call
```

Comment:
It does seem hard to trace data through the queue, but I'm not sure if we can do anything about it at this point without a large refactor. @pmorie @vaikas-google what are your thoughts RE cleaning up the flow? Either way, I think that work would be outside the scope.